v2/yarpctchannel/outbound.go

```diff
@@ -177,16 +176,16 @@ func callWithPeer(ctx context.Context, req *yarpc.Request, reqBody *yarpc.Buffer
 			"does not match the service name received in the response: sent %q, got: %q", req.Service, resSvcName)
 	}
 
-	var appErr error
+	var appErrInfo *yarpcerror.Info
 	if res.ApplicationError() {
 		// TODO(mhp): this is a filler error for now to preserve current yarpc
 		// error behavior. This should later be more well-defined.
-		appErr = errors.New("application error")
+		appErrInfo = &yarpcerror.Info{}
```

Comment:
should this have `CodeUnkown` and message set to "application error"?

Reply:
I hope `CodeUnknown` is implied by the zero object.

Reply:
Unfortunately no, the default for `Code` is `CodeOk`